crates/types/src/config/cli_option_overrides.rs

```diff
@@ -74,10 +75,21 @@ pub struct CommonOptionCliOverride {
     #[clap(long, env = "RESTATE_CLUSTER_NAME", global = true)]
     pub cluster_name: Option<String>,
 
-    /// If true, then a new cluster is bootstrapped. This node *must* have an admin
-    /// role and a new nodes configuration will be created that includes this node.
-    #[clap(long, global = true)]
-    pub allow_bootstrap: Option<bool>,
+    /// Auto Cluster Provisioning
+    ///
+    /// If true, then this node is allowed to automatically provision as a new cluster.
+    /// This node *must* have an admin role and a new nodes configuration will be created that includes this node.
+    ///
+    /// auto-provision is allowed by default in development mode and is disabled if restate-server runs with `--production` flag
+    /// to prevent cluster nodes from forming their own clusters, rather than forming a single cluster.
+    ///
+    /// Use `restatectl` to provision the cluster/node if automatic provisioning is disabled.
+    ///
+    /// This can also be explicitly disabled by setting this value to false.
+    ///
+    /// Default: true
+    #[clap(long, global = true, alias = "allow_bootstrap")]
+    pub auto_provision: Option<bool>,
```

Comment:
This is a backward incompatible change. I am sure you are aware of this. But won't this mean that if I updated the binary and reused my configuration files, suddenly the `auto_provision` will be set to `true` while I have `allow-bootstrap = false` in my (old) config file?